aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -495,6 +464,28 @@ impl AptosVM {
         }
     }
 
+    fn charge_change_set_and_respawn_session<'r, 'l>(
+        &'l self,
+        session: SessionExt,
+        resolver: &'r impl MoveResolverExt,
+        gas_meter: &mut impl AptosGasMeter,
+        change_set_configs: &ChangeSetConfigs,
+        txn_data: &TransactionMetadata,
+    ) -> Result<RespawnedSession<'r, 'l>, VMStatus> {
+        let change_set_ext = session.finish(&mut (), change_set_configs)?;
+        gas_meter.charge_io_gas_for_write_set(change_set_ext.write_set().iter())?;
+        gas_meter.charge_storage_fee_for_all(
+            change_set_ext.write_set().iter(),
+            change_set_ext.change_set().events(),
+            txn_data.transaction_size,
+            txn_data.gas_unit_price,
+        )?;
+
+        // TODO(Gas): Charge for aggregator writes
+        let session_id = SessionId::txn_meta(txn_data);
+        RespawnedSession::spawn(&self.0, session_id, resolver, change_set_ext)
```

Comment:
Maybe I missed something, but here we're still spawning a new (Move VM) Session, right? Could you further elaborate on exactly what duplicated work this PR saves?

Reply:
It doesn't create one less session (except for the multi-sig flow, where yet another session was created). It removes duplicaton in code -- `charge_storage_fee_for_all()` for example is now called at only one place instead of 3.

Reply:
Ah now I see what you mean by "removing the duplication". Agreed that it makes sense to define a session at the Aptos VM level that stores and provides all the info one needs to execute a transaction, even if the Move VM stops doing this in the future.